console/console-init/ui/src/modules/iot-device/DeviceListPage.tsx

```diff
@@ -64,6 +64,13 @@ export default function DeviceListPage() {
   const createDeviceFormLink = `/iot-projects/${namespace}/${projectname}/devices/addform`;
   const createDeviceJsonLink = `/iot-projects/${namespace}/${projectname}/devices/addjson`;
 
+  const setSearchParam = useCallback(
```

Comment:
can you try this `useSearchParamsPageChange` custorm hook for changing page value.